sdk/src/androidTest/java/com/mapbox/maps/MapIntegrationTest.kt

```diff
@@ -480,6 +490,105 @@ class MapIntegrationTest {
     }
   }
 
+  @Test
+  fun testApplyLargeDataAfterCreate() {
+    countDownLatch = CountDownLatch(1)
+    rule.scenario.onActivity {
+      mapView = MapView(it)
+      mapboxMap = mapView.getMapboxMap()
+      it.frameLayout.addView(mapView)
+      mapboxMap.setCamera(
+        CameraOptions.Builder()
+          .center(Point.fromLngLat(0.0, 0.0))
+          .zoom(9.0)
+          .build()
+      )
+      mapboxMap.loadStyleUri(Style.MAPBOX_STREETS) { style ->
+        // load a large feature collection in constructor
+        val source = geoJsonSource("source") {
+          featureCollection(getFeatureCollection())
+        }
+        // change geometry immediately after create the source
+        source.geometry(Point.fromLngLat(0.0, 0.0)) {}
+
+        style.addSource(source)
+        style.addLayer(
+          circleLayer("layer", "source") {
+            circleColor("red")
+            circleRadius(10.0)
+          }
+        )
+        mapView.postDelayed(
+          {
+            mapboxMap.queryRenderedFeatures(
+              ScreenCoordinate(mapView.width / 2.0, mapView.height / 2.0),
+              RenderedQueryOptions(listOf("layer"), null)
+            ) { result ->
+              if (result.value?.size == 1) {
+                countDownLatch.countDown()
+              }
+            }
+          },
+          5_000L
+        )
+      }
+    }
+    if (!countDownLatch.await(10, TimeUnit.SECONDS)) {
+      throw TimeoutException()
+    }
+  }
+
+  @Test
+  fun testApplyDataAfterLargeData() {
+    countDownLatch = CountDownLatch(1)
+    rule.scenario.onActivity {
+      mapView = MapView(it)
+      mapboxMap = mapView.getMapboxMap()
+      it.frameLayout.addView(mapView)
+      mapboxMap.setCamera(
+        CameraOptions.Builder()
+          .center(Point.fromLngLat(0.0, 0.0))
+          .zoom(2.0)
+          .build()
+      )
+      mapboxMap.loadStyleUri(Style.MAPBOX_STREETS) { style ->
+        val source = geoJsonSource("source") {
+          data("")
+        }
+        // load a large feature collection
+        source.featureCollection(getFeatureCollection())
+        // change geometry immediately
+        source.geometry(Point.fromLngLat(0.0, 0.0)) {}
```

Comment:
Comment seems to be confusing as we're applying not immediately here but in async manner after large `featureCollection` applied in a sync manner.